src/ArrowFunctions.cpp

```diff
@@ -1078,31 +1084,88 @@ int cpp_writeMultiColToParquet(const char* filename, void* column_names,
           auto data_ptr = (uint8_t*)ptr_arr[i];
           parquet::ByteArrayWriter* ba_writer =
           static_cast<parquet::ByteArrayWriter*>(rg_writer->NextColumn());
-          int64_t count = 0;
-          int64_t byteIdx = 0;
+          if (objType_ptr[i] == SEGARRAY) {
+            int64_t byteIdx = 0;
 
-          // identify the starting byte index
-          if (x > 0){
-            byteIdx = idxQueue_str.front();
-            idxQueue_str.pop();
-          }
-          
-          while(count < batchSize) {
-            parquet::ByteArray value;
-            int16_t definition_level = 1;
-            value.ptr = reinterpret_cast<const uint8_t*>(&data_ptr[byteIdx]);
-            int64_t nextIdx = byteIdx;
-            while (data_ptr[nextIdx] != 0x00){
-              nextIdx++;
+            // identify the starting byte index
+            if (x > 0){
+              byteIdx = idxQueue_str.front();
+              idxQueue_str.pop();
+            }
+            
+            auto offset_ptr = (int64_t*)offset_arr[i];
+            int64_t offIdx = 0; // index into offsets
+
+            if (x > 0){
+              offIdx = idxQueue_segarray.front();
+              idxQueue_segarray.pop();
+            }
```

Comment:
could we combine the two `if (x > 0)` cases to set `byteIdx` and `offIdx` at the same time?
```chapel
auto offset_ptr = (int64_t*)offset_arr[i];
int64_t byteIdx = 0;
int64_t offIdx = 0; // index into offsets
// identify the starting byte and offset index
if (x > 0){
  byteIdx = idxQueue_str.front();
  idxQueue_str.pop();

  offIdx = idxQueue_segarray.front();
  idxQueue_segarray.pop();
}
```

Also i point this out in the push case as well, but should we be doing `byteIdx = idxQueue_str.front();` or `byteIdx = idxQueue_segarray.front();`

Reply:
Oh yeah, good idea. I think I was trying to keep them separate in my head while adding this to make sure I could track everything, but definitely no need for 2 `if` blocks.

Also, good catch on the `idxQueue` stuff. The `byteIdx` value should be pushed to `idxQueue_str` and the `offIdx` to `idxQueue_segarray`. This has been updated as well.